x-pack/plugins/apm/server/routes/alerts/rule_types/anomaly/register_anomaly_rule_type.ts

```diff
@@ -188,9 +185,9 @@ export function registerAnomalyRuleType({
                   ...termQuery('by_field_value', ruleParams.transactionType, {
                     queryEmptyString: false,
                   }),
-                  ...termQuery(
+                  ...termsQuery(
                     'detector_index',
-                    getApmMlDetectorIndex(ApmMlDetectorType.txLatency)
+                    getApmMlDetectorIndex(ruleParams.anomalyDetectorTypes)
```

Comment:
This looks incorrect. I'd imagine you'd have to do:

```suggestion
                    ruleParams.anomalyDetectorTypes.map(type => getApmMlDetectorIndex(type))
```